articles/node_lifecycle.md

```diff
@@ -27,224 +27,447 @@ Last Modified: {% if page.last_modified %}{{ page.last_modified }}{% else %}{{ p
+#### **Primary State: `Active`**
+Equivalent to the default behavior of a created `Node`.
+Thus, while in this state, the node performs any processing, responds to service requests, reads and processes data, produces output, etc.
 
-#### Valid transitions out of Inactive
+*State Assumptions / Behaviors*
+- Managed Entities are allocated
+- Events processed for these Managed Entities
 
-- A node may transition to the `Finalized` state via the `shutdown` transition.
-- A node may transition to the `Unconfigured` state via the `cleanup` transition.
-- A node may transition to the `Active` state via the `activate` transition.
+*Outbound State Transitions*
+- `Transition:DEACTIVATE` →  `Activating`
+- `Transition:SHUTDOWN` → `ShuttingDown`
+- `RAISE_ERROR` →  `ErrorProcessing`
 
-### Primary State: Active
 
-This is the main state of the node's life cycle.
-While in this state, the node performs any processing, responds to service requests, reads and processes data, produces output, etc.
+#### **Primary State: `Finalized`**
+The `Finalized` state is the state in which the node ends immediately before being destroyed.
+This state exists to support debugging and introspection.
+A node which has failed will remain visible to system introspection and may be potentially introspectable by debugging tools instead of directly destructing.
+There will be no Lifecycle internal/backend policy for automatically restarting a node.
 
-If an error that cannot be handled by the node/system occurs in this state, the node will transition to `ErrorProcessing`.
+*State Assumptions / Behaviors*
+- This state can only be reached via a `ShuttingDown` → `SUCCESS` → `Finalized`
+- Anything the node is deemed responsible for (e.g., Managed Entities) using CPU is/are stopped
+- Anything the node is deemed responsible for (e.g., Managed Entities) using RAM/Mem is/are de-allocated
+- `node->destroy()` can be safely run
 
-#### Valid transitions out of Active
 
-- A node may transition to the `Inactive` state via the `deactivate` transition.
-- A node may transition to the `Finalized` state via the `shutdown` transition.
+**Terminal state**: No Outbound Transitions
 
-### Primary State: Finalized
 
-The `Finalized` state is the state in which the node ends immediately before being destroyed.
-This state is always terminal the only transition from here is to be destroyed.
 
-This state exists to support debugging and introspection.
-A node which has failed will remain visible to system introspection and may be potentially introspectable by debugging tools instead of directly destructing.
-If a node is being launched in a respawn loop or has known reasons for cycling it is expected that the supervisory process will have a policy to automatically destroy and recreate the node.
+#### **Primary State: `UncleanFinalized`**
+This state exists to denote a node that failed recovery in `ErrorProcessing` and therefor the `LifecycleNode` cannot be guaranteed to be cleanly destroyed.
```

Comment:
Grammar and therefor (archaic) vs. _therefore_, consider
```
This state exists to denote that the node failed recovery in `ErrorProcessing` and therefore
the `LifecycleNode` cannot be guaranteed to be cleanly destroyed.
```
Also use of therefor vs. _therefore_ in the next sentence too.